File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] arr) {
        Arrays.sort(arr,(a,b)->Integer.compare(a[0],b[0]));
        ArrayList<int[]> list = new ArrayList<>();
        int[] curr = arr[0];
        list.add(curr);
        if(arr.length==1){
            return arr;
        }
        for(int i=1;i<arr.length;i++){
            curr = arr[i];
            int[] prv = list.get(list.size()-1);
            if(curr[0]<=prv[1]){
                prv[1] = Math.max(curr[1],prv[1]);
            }else{
                list.add(curr);
            }
        }
        return list.toArray(new int[list.size()][]);
    }
}

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, arr: List[List[int]]) -> List[List[int]]:
        arr.sort()
        ans = [arr[0]]
        for i in arr:
            if ans[-1][1]<i[0]:
                # val = [min(arr[i-1][0],arr[i][0]),max(arr[i-1][1],arr[i][1])]
                ans.append(i)
            else:
                ans[-1][1] = max(i[1],ans[-1][1])
        return ans
        